sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -298,6 +309,28 @@ trait FileFormat {
 }
 
 /**
+ * The base class file format that is based on text file.
+ */
+abstract class TextBasedFileFormat extends FileFormat {
+  private var codecFactory: CompressionCodecFactory = null
+  override def isSplitable(
+      sparkSession: SparkSession,
+      options: Map[String, String],
+      path: Path): Boolean = {
+    if (codecFactory == null) {
+      synchronized {
+        if (codecFactory == null) {
+          codecFactory = new CompressionCodecFactory(
```

Comment:
We need to pass `options` into `CompressionCodecFactory` here? It seems `getCodec` only checks suffixes in `input`.
https://github.com/apache/hadoop-common/blob/HADOOP-3628/src/core/org/apache/hadoop/io/compress/CompressionCodecFactory.java#L153